src/main/java/com/udojava/evalex/Expression.java

```diff
@@ -1836,6 +1840,57 @@ public String getExpression() {
 		return result;
 	}
 
+	/**
+	 * Verify the validity of the variables
+	 * if expression has undefined vars or circular reference vars throw ExpressionException
+	 * @return void
+	 */
+	public void varDefValidation() {
+		final String mark = "+";
+		Queue<String> checks = new LinkedList<String>(getUsedVariables());
+		checks.offer(mark);
+		final Set<String> varWithValue = new HashSet<String>();
+		int lastvarWithValueCnt = varWithValue.size();
+        String lastCheck = "";
+        final Set<String> lasttodocheck = new HashSet<String>();
+		while (!checks.isEmpty()) {
+			String check = checks.peek();
+			if (mark.equals(check)) {
+				if (varWithValue.size() < lastvarWithValueCnt ) {
+					throw new ExpressionException("circular reference var");
```

Comment:
If available, you should always include the name of the variable which is causing the problem.